src/services/config.js

```diff
@@ -110,19 +120,24 @@ class ConfigData {
     }
   }
 
-  addProfile(id, accountSid, region) {
-    // Clean all the inputs.
+  addProfile(id, accountSid, region, apiKey, apiSecret) {
+    //  Clean all the inputs.
     id = this.sanitize(id);
     accountSid = this.sanitize(accountSid);
     region = this.sanitize(region);
 
     const existing = this.getProfileById(id);
+
+    //  Update the historical projects array
     if (existing) {
       existing.accountSid = accountSid;
       existing.region = region;
     } else {
-      this.projects.push(new ConfigDataProfile(id, accountSid, region));
+      this.projects.push(new ConfigDataProject(id, accountSid, region));
     }
```

Comment:
I don't think we should be adding/updating `this.projects`. All updated profiles should be removed from `this.projects` and added to `this.profiles`. New profiles should just be going in `this.profiles`.